GitCommands/Git/GitModule.cs

```diff
@@ -633,14 +633,12 @@ public string RunCacheableCmd(string cmd, string arguments = "", Encoding encodi
             }
 
             byte[] cmdout, cmderr;
-            if (GitCommandCache.TryGet(arguments, out cmdout, out cmderr))
+            if (!GitCommandCache.TryGet(arguments, out cmdout, out cmderr))
             {
-                return StripAnsiCodes(EncodingHelper.DecodeString(cmdout, cmderr, ref encoding));
-            }
-
-            GitCommandHelpers.RunCmdByte(cmd, arguments, WorkingDir, null, out cmdout, out cmderr);
+                GitCommandHelpers.RunCmdByte(cmd, arguments, WorkingDir, null, out cmdout, out cmderr);
 
-            GitCommandCache.Add(arguments, cmdout, cmderr);
+                GitCommandCache.Add(arguments, cmdout, cmderr);
+            }
```

Comment:
to me this looks the inversion of what it said to achieve - the original had less nesting and early return

Reply:
The intended improvement here was the removal of the duplicated `return` line. Note this was part of a secondary commit titled "Invert condition to remove duplicated return expression".

Reply:
ahh, right, I see it now.
I looked at the combined diff.